py2puml/inspection/inspectclass.py

```diff
@@ -119,34 +78,42 @@ def parse_class_type(
                     attr_type = f"{composition_rel}[{', '.join(component_names)}]"
                 else:
                     attr_type = attr_raw_type
-            uml_attr = UmlAttribute(attr_name, attr_type)
+            uml_attr = UmlAttribute(attr_name, attr_type, True)
             definition_attrs.append(uml_attr)
 
-    handle_inheritance_relation(class_type, class_type_fqdn, root_module_name, domain_relations)
-
+    return definition_attrs
 
-def parse_type(
-    definition_type: Type,
+def inspect_class_type(
+    class_type: Type,
+    class_type_fqdn: str,
     root_module_name: str,
     domain_items_by_fqdn: Dict[str, UmlItem],
     domain_relations: List[UmlRelation]
 ):
-    definition_type_fqdn = f'{definition_type.__module__}.{definition_type.__name__}'
-    if definition_type_fqdn not in domain_items_by_fqdn:
-        if issubclass(definition_type, Enum):
-            parse_enum_type(definition_type, definition_type_fqdn, domain_items_by_fqdn)
-        elif getattr(definition_type, '_fields', None) is not None:
-            parse_namedtupled_class(definition_type, definition_type_fqdn, domain_items_by_fqdn)
-        else:
-            parse_class_type(definition_type, definition_type_fqdn, root_module_name, domain_items_by_fqdn, domain_relations)
+    attributes = inspect_static_attributes(
+        class_type, class_type_fqdn, root_module_name,
+        domain_items_by_fqdn, domain_relations
+    )
+    instance_attributes, compositions = parse_class_constructor(class_type, class_type_fqdn, root_module_name)
+    attributes.extend(instance_attributes)
+    domain_relations.extend(compositions.values())
 
+    handle_inheritance_relation(class_type, class_type_fqdn, root_module_name, domain_relations)
 
-def parse_item_module(
-    domain_item_module: ModuleType,
+def inspect_dataclass_type(
+    class_type: Type[dataclass],
+    class_type_fqdn: str,
     root_module_name: str,
     domain_items_by_fqdn: Dict[str, UmlItem],
     domain_relations: List[UmlRelation]
 ):
-    # processes only the classes imported and defined within the given root module
-    for definition_type in filter_domain_definitions(domain_item_module, root_module_name):
-        parse_type(definition_type, root_module_name, domain_items_by_fqdn, domain_relations)
+    for attribute in inspect_static_attributes(
+        class_type,
+        class_type_fqdn,
+        root_module_name,
+        domain_items_by_fqdn,
+        domain_relations
+    ):
+        attribute.static = False
```

Comment:
Why do you need to remove staticity here?

Reply:
because the @dataclass decorator creates a constructor and instance attributes from the static class attributes. Therefore, the static attribute definitions document the instance attributes, which are not static anymore.